frontstage/views/sign_in.py

```diff
@@ -29,27 +29,50 @@ def login():
-                                      client_secret=app.config['RAS_FRONTSTAGE_CLIENT_SECRET'])
-
+            token_url = app.config['ONS_OAUTH_PROTOCOL'] + app.config['ONS_OAUTH_SERVER'] + app.config['ONS_TOKEN_ENDPOINT']
+            data = {
+                'grant_type': 'password',
+                'client_id': app.config['RAS_FRONTSTAGE_CLIENT_ID'],
+                'client_secret': app.config['RAS_FRONTSTAGE_CLIENT_SECRET'],
+                'username': username,
+                'password': password,
+            }
+            headers = {
+                'Accept': 'application/json',
+                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+            }
+
+            oauth2_response = requests.post(url=token_url, data=data , headers=headers, auth=(app.config['RAS_FRONTSTAGE_CLIENT_ID'],
+                                                                                              app.config['RAS_FRONTSTAGE_CLIENT_SECRET']))
+            oauth2_token = json.loads(oauth2_response.text)
+            # Check to see that this user has not attempted to login too many times or that they have not forgot to
+            # click on the activate account URL in their email by checking the error message back from the OAuth2 server
+            if oauth2_response.status_code == 401:
+                oauth2Error = json.loads(oauth2_response.text)
+                if oauth2Error['detail'] == 'Unauthorized user credentials':
+                    return render_template('sign-in/sign-in.html', _theme='default', form=form, data={"error": {"type": "failed"}})
+                elif 'User account locked' in oauth2Error['detail']:
+                    logger.warning('User account is locked on the OAuth2 server')
+                    return render_template('sign-in/sign-in.trouble.html', _theme='default', form=form,
+                                           data={"error": {"type": "account locked"}})
+                elif 'User account not verified' in oauth2Error['detail']:
+                    logger.warning('User account is not verified on the OAuth2 server')
+                    return render_template('sign-in/sign-in.account-not-verified.html', _theme='default', form=form,
+                                           data={"error": {"type": "account not verified"}})
+                else:
+                    logger.error('OAuth 2 server generated 401 which is not understood', oauth2error=oauth2Error['detail'])
+                    return render_template('sign-in/sign-in.html', _theme='default', form=form,
+                                           data={"error": {"type": "failed"}})
+            if oauth2_response.status_code != 201:
+                logger.error('Unknown error from the OAuth2 server', ouath2_response=oauth2_response.txt,  status_code=oauth2_response.status_code)
+                raise ExternalServiceError(oauth2_response.text)
```

Comment:
ExternalServiceError takes a response object so this should be `ExternalServiceError(oauth2_response)`